argoverse/utils/centerline_utils.py

```diff
@@ -101,24 +101,32 @@ def centerline_to_polygon(
 
 
 def convert_lane_boundaries_to_polygon(right_lane_bounds: np.ndarray, left_lane_bounds: np.ndarray) -> np.ndarray:
-    """
-    Take a left and right lane boundary and make a polygon of the lane segment, closing both ends of the segment.
+    """Given 3d left and right boundaries of a lane segment, provide the exterior vertices of the 2d or 3d lane segment polygon.
+
+    Note: We chain the right segment with a reversed left segment, and then repeat the first vertex. In other words, 
+    the first and last vertex are identical.
 
-    These polygons have the last vertex repeated (that is, first vertex == last vertex).
+    L _________
+      .       .
+      .       .
+    R _________
 
     Args:
-       right_lane_bounds: Right lane boundary points. Shape is (N, 2).
-       left_lane_bounds: Left lane boundary points.
+       right_lane_bounds: Array of shape (K,2) or (K,3) representing right lane boundary points.
+       left_lane_bounds: Array of shape (M,2) or (M,3) representing left lane boundary points.
 
     Returns:
-       polygon: Numpy array of shape (2N+1,2)
+       polygon: Numpy array of shape (K+M+1,2) or (K+M+1,3)
     """
-    assert right_lane_bounds.shape[0] == left_lane_bounds.shape[0]
+    # assert right_lane_bounds.shape[0] == left_lane_bounds.shape[0]
```

Comment:
We need to check that the last dimension of `left_lane_bounds` and `right_lane_bounds` match up right?

Reply:
that's true, i can add that check. before we didn't realize boundaries, so left and right bounds were purely inferred, so they always match in the first dimension, also, previously